### PR TITLE
Move to import house contacts from the Clerk XML feed

### PR DESCRIPTION
**Pull request by joelcollinsdc**
From #276, the Clerk of the House provides XML for member information.  There are several name fields in the XML that the Clerk provides that do not directly match the name fields in legislators-current.

### CODE REVIEW

File: scripts/house_contacts.py [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
 	today = datetime.now().date()
 
-	# default to not caching
-	cache = utils.flags().get('cache', False)
-	force = not cache
-
 	y = load_data("legislators-current.yaml")
 
+	# TODO use download util?
+	xml = requests.get("http://clerk.house.gov/xml/lists/MemberData.xml")
```
Comment (konklone) [inline scripts/house_contacts.py]: Could really use an HTTPS URL for this...

**Comment by konklone** [comment]
@joelcollinsdc Mind if we give you write permissions for the repo? You're definitely at the point where we'd welcome you submitting PRs that go branch-to-branch instead of fork-to-branch. It makes it easier for us to hop in and tweak stuff before merging, too.

**Comment by konklone** [comment]
To document for the PR (since the changes are too big to show on the `Files changed` tab), this adds a `list` and `formal` field for each member, like so:

```diff
@@ -6163,6 +6219,8 @@
     middle: R.
     last: Carter
     official_full: John R. Carter
+    list: Carter, John
+    formal: Mr. Carter of Texas
```

Some of the `formal` names have `of [state]` at the end, some don't. @joelcollinsdc, what determines whether they do or not?

@joelcollinsdc: did this commit changing Paul Ryan's name come from the script, or from a manual override?

```diff
@@ -26516,7 +26516,8 @@
   name:
     first: Paul
     last: Ryan
-    official_full: Paul Ryan
+    official_full: Paul D. Ryan
+    middle: D.
```

File: scripts/house_contacts.py (hunk 1 above)
Comment (aboodman) [inline scripts/house_contacts.py]: heh

**Comment by joelcollinsdc** [comment]
Sure!   I'll keep any stuff out of master unless told otherwise.

**Comment by konklone** [comment]
Yeah, in general PRs are always preferred for any changes to the repo, to keep visibility high. I sent you an invite to the organization.

File: scripts/house_contacts.py (hunk 1 above)
Comment (joelcollinsdc) [inline scripts/house_contacts.py]: no HTTPS version of the clerk URL works at the moment unfortunately.  I'll look into that though.

**Comment by joelcollinsdc** [comment]
@konklone I can check with the Clerk's office on this, from what I understand the Member fills out their preference for their names.  The "D." in "Paul D. Ryan" was an official request, and came from the source XML, no hand editing was done.

**Comment by konklone** [comment]
:+1: Looks good to me.

**Comment by JoshData** [comment]
Hang on there. I un-did the merge. This needs a little more work...

* This doesn't handle nicknames correctly (`middle: C. "Hank"`). Every rep with an existing nickname field needs to be reviewed (e.g. see `Andy Barr`).
* The change in the first name of Scott Rigel (`Edward` => `E.`) is probably going to mess up the name display logic on GovTrack due to his record previously having `Scott` be both his middle name and nickname.
* I'm not sure I'd want to add in the `formal` name. It's House-specific, it's hard to imagine a use case for users of this data, and it's also time-specific (it has e.g. ` of Texas` on names that need disambiguation, which of course only makes sense if you're referring to that person as of the time of the last scrape --- it would be wrong to use that name in conjunction with any historical events, e.g. sponsorship of a bill at a time when the name wasn't ambiguous).
* `list` is basically a duplication of other data fields, it's House-specific, and it isn't in any of the historical records --- so here too I'm not sure I'd want to add this.
* That's mostly also true of `official_full`. I had added `official_full` originally as a sanity check for the current house contacts scraper (because the source data lacked bioguide IDs, iirc). With this PR we don't need that sanity check anymore, so we could consider dropping the field.

**Comment by joelcollinsdc** [comment]
I moved the house-clerk-xml branch into this repo.  Since this PR was closed anyways I'll resubmit and work on it there.

**Pull request merged**